agent/07_agent_add_node.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/network.sh
+source $SCRIPTDIR/ocp_install_env.sh
+source $SCRIPTDIR/utils.sh
+source $SCRIPTDIR/validation.sh
+source $SCRIPTDIR/agent/common.sh
+
+early_deploy_validation
+
+function build_node_joiner() {
+  # Build installer
+  pushd .
+  cd $OPENSHIFT_INSTALL_PATH
+  TAGS="${OPENSHIFT_INSTALLER_BUILD_TAGS:-libvirt baremetal}" DEFAULT_ARCH=$(get_arch) hack/build-node-joiner.sh
+  popd
+}
+
+function approve_csrs() {
+  while true; do
+    pending_csrs=$(oc get csr | grep Pending)
+    if [[ ${pending_csrs} != "" ]]; then
+      echo "Approving CSRs: $pending_csrs"
+      echo $pending_csrs | cut -d ' ' -f 1 | xargs oc adm certificate approve
+    fi
+    sleep 10
+  done
+}
+
+if [ -z "$KNI_INSTALL_FROM_GIT" ]; then
+  # Extract node-joiner from the release image
+  baremetal_installer_image=$(oc adm release info --image-for=baremetal-installer --registry-config=$PULL_SECRET_FILE)
+  container_id=$(podman create --authfile $PULL_SECRET_FILE ${baremetal_installer_image} ls)
+  podman start $container_id
+  podman export $container_id > baremetal-installer.tar
+  tar xf baremetal-installer.tar usr/bin/node-joiner
+  cp usr/bin/node-joiner $OCP_DIR/node-joiner
+  rm -rf usr baremetal-installer.tar
+  podman rm $container_id
+else
+  # Build the node-joiner from source
+  build_node_joiner
+  cp "$OPENSHIFT_INSTALL_PATH/bin/node-joiner" "$OCP_DIR"
+fi
+
+rm $OCP_DIR/add-node/.openshift_install_state.json | true
+
+get_static_ips_and_macs
```

Comment:
TBTH it seems to me that `AGENT_EXTRAWORKER_NODE_TO_ADD` is not necessary, but yet another variable to be configured that doesn't seem really needed (I'm usually keen to avoid adding other config var if not strictly necessary).
In our case it seems to me that it would be sufficient to use just the already existing `NUM_EXTRA_WORKERS` var in the step 07 (and that should probably help in keeping the funcs in common.sh also, considering also that one step is invoked after the other)

Reply:
> The reason get_static_ips_and_macs is moved to common.sh and instead of leaving it in 05_agent_configure.sh is that function is called here so that AGENT_EXTRA_WORKER_IPS can be used along with AGENT_EXTRAWOKER_NODE_TO_ADD to fetch the correct IP address for the node being added.

I carefully reviewed its usage, and still it does not seem correct to me applying twice in the same workflow, also because the function has some side effects. That method is really part of just step 5, and should stay there. 
The only bit of info required by step 7 is `AGENT_EXTRA_WORKERS_IPS`, and such var should be already be available as calculated by step 5 during the workflow